src/pypdfium2/_helpers/textinserter.py

```diff
@@ -0,0 +1,96 @@
+# SPDX-FileCopyrightText: 2022 Anurag Bansal <anurag.bansal.585@gmail.com>
+# SPDX-FileCopyrightText: 2022 geisserml <geisserml@gmail.com>
+# SPDX-License-Identifier: Apache-2.0 OR BSD-3-Clause
+
+import ctypes
+from pypdfium2 import _pypdfium as pdfium
+from pypdfium2._helpers.opener import open_page
+
+try:
+    import uharfbuzz as harfbuzz
+except ImportError:
+    have_harfbuzz = False
+else:
+    have_harfbuzz = True
+
+
+def insert_text(
+        pdf,
+        page_index,
+        text,
+        pos_x,
+        pos_y,
+        font_path,
+        font_size,
+        font_type,
+        font_is_cid,
+    ):
+    """
+    Insert text into a PDF page at a specified position. This function supports Asian scripts such as Hindi.
```

Comment:
The comment can probably be - "Insert text into a PDF page at a specified position using the language's ligature (https://en.wikipedia.org/wiki/Ligature_(writing))", as it's not just re-arrangement which is happening since these languages usually combine two characters to form a entirely new character (usually called as glyphs, and identified using cid - character id) which in themselves do not have a unicode value assigned to them.

Maybe even better explanation can be added then the one I provided.

Reply:
Yes, that sounds better. I'll add it.

Reply:
I now wrote `using the writing system's ligature`, as the code only gets the characters and doesn't know which language it is.